core/src/main/java/de/jplag/Submission.java

```diff
@@ -314,6 +285,7 @@ public Submission copy() {
         Submission copy = new Submission(name, submissionRootFile, isNew, files, language);
         copy.setTokenList(new ArrayList<>(tokenList));
         copy.setBaseCodeComparison(baseCodeComparison);
+        copy.state = state;
```

Comment:
We should either use setters or direct access :)

Reply:
State should not be settable from the outside, so this would mean I would create a private setter just for this copy method. The others (token list, base code comparison) already have public setters anyway.